presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -183,9 +189,37 @@ private Expression removeAllDynamicFilters(Expression expression)
         {
             DynamicFilters.ExtractResult extractResult = extractDynamicFilters(expression);
             if (extractResult.getDynamicConjuncts().isEmpty()) {
-                return expression;
+                return removeNestedDynamicFilters(expression);
             }
-            return combineConjuncts(metadata, extractResult.getStaticConjuncts());
+            return combineConjuncts(metadata, extractResult.getStaticConjuncts()
+                    .stream()
+                    .map(this::removeNestedDynamicFilters)
+                    .collect(toImmutableList()));
+        }
+
+        private Expression removeNestedDynamicFilters(Expression expression)
+        {
+            return ExpressionTreeRewriter.rewriteWith(new RemoveDynamicFilters(), expression);
+        }
+    }
+
+    private static class RemoveDynamicFilters
+            extends io.prestosql.sql.tree.ExpressionRewriter<Void>
+    {
+        @Override
+        public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            LogicalBinaryExpression rewrittenNode = treeRewriter.defaultRewrite(node, context);
+
+            ImmutableList.Builder<Expression> expressionBuilder = ImmutableList.builder();
+            if (!isDynamicFilter(rewrittenNode.getLeft())) {
+                expressionBuilder.add(rewrittenNode.getLeft());
+            }
+            if (!isDynamicFilter(rewrittenNode.getRight())) {
+                expressionBuilder.add(rewrittenNode.getRight());
+            }
+
```

Comment:
return original node if nothing changes (both expressions are not dynamic filters).

Reply:
done